CRM/Emailfiling/Helper/MailAccountSettings.php

```diff
@@ -93,79 +93,83 @@ public function isEnabled() {
    * @param int $value
    *   New value for the field.
    */
-  public function change($value) {
+  public function toggle($value) {
     // Since setting contains array of field values, we can't save new value
     // to it directly, so let's load array first.
     $setting = \Civi::settings()->get($this->settingName);
+    $accountId = $this->getAccountId();
 
     // If we need to disable the feature - let's just remove respective value
     // from array.
     if (!$value) {
-      if (isset($setting[$this->accountId])) {
-        unset($setting[$this->accountId]);
+      if (isset($setting[$accountId])) {
+        unset($setting[$accountId]);
       }
     }
     // Add new value to array for current mail account.
     else {
-      $setting[$this->accountId] = $value ? 1 : 0;
+      $setting[$accountId] = $value ? 1 : 0;
     }
 
     // Save updated array.
     \Civi::settings()->set($this->settingName, $setting);
   }
 
   /**
-   * Loads mail account settings.
+   * Loads mail account settings by it's id.
    *
-   * This function can load account by id or by email address.
-   * It may be useful to get mail account id and ensure it actually exists.
+   * @param int $accountId
+   *   Mail account id.
    *
    * @throws \CiviCRM_API3_Exception
    *   An exception generated by civicrm_api3().
    */
-  private function loadAccountSettings() {
-    // Load settings by id if available.
-    if (!empty($this->accountId)) {
-      $result = civicrm_api3('MailSettings', 'get', [
-        'sequential' => 1,
-        'is_default' => 0,
-        'id' => $this->accountId,
-      ]);
-
-      if (!empty($result['values'][0])) {
-        $this->account = $result['values'][0];
-        $this->accountId = $this->account['id'];
-        return;
-      }
-    }
-
-    if (empty($this->email)) {
+  private function loadAccountById($accountId) {
+    if (empty($accountId)) {
       return;
     }
 
-    // Load settings by email address.
     $result = civicrm_api3('MailSettings', 'get', [
       'sequential' => 1,
       'is_default' => 0,
-      'domain' => $this->email->host,
-      'username' => $this->email->mailbox,
+      'id' => $accountId,
     ]);
 
-    if (!empty($result['values'][0])) {
-      $this->account = $result['values'][0];
-      $this->accountId = $this->account['id'];
+    $this->account = $result['values'][0] ?? NULL;
+  }
+
+  /**
+   * Loads mail account settings by email.
+   *
+   * This function can load account by id or by email address.
+   * It may be useful to get mail account id and ensure it actually exists.
+   *
+   * @param Mail_RFC822 $email
+   *   Email address that mail account has.
+   *
+   * @throws \CiviCRM_API3_Exception
+   *   An exception generated by civicrm_api3().
+   */
+  private function loadAccountByEmail(Mail_RFC822 $email) {
+    if (empty($email)) {
```

Comment:
This would be an object and will not be empty. Except if we want to check if a particular property exists in the object or not.

e.g
```php
if (empty($email->mailbox) {
return
}
```

or we might not even need this condition at all.